### PR TITLE
clear timeouts/intervals after each test

### PR DESCRIPTION
**Pull request by Bjvanminnen**
I was writing some tests for some async code that was giving me trouble because the code I was testing would fire off a `setTimeout` that would get executed after the code finished.

I realized that solving this was actually pretty easy - it requires that we track all creations, and then clear them when the test finishes.

Validated that with code like the following, we no longer see "set timeout called"
```js
describe('something', () => {
  it('calls an async action in a sync test', () => {
    assert(true);
    setTimeout(() => console.log('set timeout called'), 1000);
  });

  it('has an async test as well', done => {
    // without changes, we see console.log from previous test while running this test.
    setTimeout(done, 2000);
  });
});
```

In addition, I ran
`npm run test`
`npm run test:unit`
`grunt karma:entry --entry=./test/integration/levelTests.js`
`grunt storybookTest`

and all were successful

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
To do that, we'd presumably need to track clearTimeout/clearInterval so we know which aren't already cleared (not too difficult).

**Comment by Bjvanminnen** [comment]
For reasons I haven't been able to understand, this was consistently failing in circle in the same spot (thought passing locally). I added some logic to also track clearTimeout/clearInterval, and only call those that hadn't been cleared on finish, in part to help debug what might be happening. Adding that logic has resulted in my tests passing consistently.

The additional changes aren't big, but prob worth a quick review before I before. PTAL when you have a chance @islemaster. Thanks.

**Comment by islemaster** [comment]
Looking now.  This is super weird - double-clearing timeouts is supposed to be safe, and I'm pretty sure we do that elsewhere.

**Comment by Bjvanminnen** [comment]
It's also possible that it was just a timing thing, and this added code changes the timing?

File: apps/test/util/testUtils.js [hunk 1]
```diff
@@ -341,3 +341,60 @@ export function restoreOnWindow(key) {
...
+    const result = setIntervalNative(...args);
+    intervalList.push(result);
+    return result;
+  };
+
+  window.clearTimeout = id => {
+    const index = timeoutList.indexOf(id);
+    if (index !== -1) {
+      timeoutList.splice(index, 1);
+     }
```
Comment (islemaster) [inline apps/test/util/testUtils.js]: Nit: Looks like this closing brace is misaligned - also in the next function.

File: apps/test/util/testUtils.js [hunk 2]
```diff
@@ -341,3 +341,60 @@ export function restoreOnWindow(key) {
...
+    const index = intervalList.indexOf(id);
+    if (index !== -1) {
+      intervalList.splice(index, 1);
+     }
+    return clearIntervalNative(id);
+  };
+
+  afterEach(() => {
+    // Note: This will end up clearing intervals/timeouts that have already
+    // been cleared
```
Comment (islemaster) [inline apps/test/util/testUtils.js]: This comment is no longer accurate.

**Pull request merged**